pkg/kubelet/status/manager.go

```diff
@@ -150,6 +154,34 @@ func (m *manager) SetPodStatus(pod *api.Pod, status api.PodStatus) {
 	m.updateStatusInternal(pod, status)
 }
 
+func (m *manager) ReconcilePodStatus(pod *api.Pod) {
+	m.podStatusesLock.Lock()
+	defer m.podStatusesLock.Unlock()
+
+	// The pod could be a mirror pod, so we should translate first.
+	uid := m.podManager.TranslatePodUID(pod.UID)
+	status, found := m.podStatuses[uid]
+	if !found {
+		// NOTE: (random-liu) What if no status? Is there any chance that this would happen?
+		glog.Warningf("Container to be reconciled has been removed in status cache: %q", format.Pod(pod))
+		return
+	}
+	if isStatusEqual(&status.status, &pod.Status) {
+		// If the status from the source is the same with the cached status,
+		// reconcile is not needed. Just return.
+		return
+	}
+	if m.needsUpdate(pod.UID, status) {
+		// The pod is going to be updated, the conflict will be reconciled after updating
+		return
+	}
+	// We should trigger an reconcile here
+	// After deleting the corresponding apiStatusVersions, the pod status will be updated in next syncBatch
+	// NOTE: (random-liu) Is apiStatusVersions-1 or status version +1 better?
+	delete(m.apiStatusVersions, pod.UID)
```

Comment:
As discussed offline, I think this should be done through a channel (apiStatusVersions is not thread-safe)